abi/encode.go

```diff
@@ -178,6 +179,19 @@ func encodeAddress(v reflect.Value) ([]byte, error) {
 	if v.Kind() == reflect.Array {
 		v = convertArrayToBytes(v)
 	}
+	if v.Kind() == reflect.String {
+		strValue := v.String()
```

Comment:
There are methods available already to parse an address from a string:
```
var addr web3.Address
if err := addr.UnmarshalText([]byte(v.String())); err != nil {
    return nil, err
}
v = addr.Bytes()
```

Reply:
resolved using `web3.HexToAddress`. For some reason that function does not return an error though, which seems weird.